internal/pkg/client/request.go

```diff
@@ -189,13 +199,24 @@ func (r *Request) nextListener() *ResponseListener {
 
 // invokeListeners if all "waitingFor" requests are done.
 func (r *Request) invokeListeners() {
+	ctx, cancel := context.WithCancel(r.Context())
+	r.cancelListeners = cancel
+
 	for {
 		// Invoke next listener if present
 		listener := r.nextListener()
 		if listener != nil {
 			listener.Invoke(r.Response)
 		} else {
-			break
+			return
```

Comment:
tu je jedno ci je `brake` alebo `return`, ... dal som tu `return ` aby bolo jasne, ze to ukoncuje beh metody.